EIPS/eip-6617.md

```diff
@@ -14,18 +14,122 @@ created: 2023-02-27
 
 This EIP offers a standard for building a bit-based permission and role system. Each permission is represented by a single bit. By using an `uint256`, up to $256$ permissions and $2^{256}$ roles can be defined. We are able to specify the importance of each permission based on the order of the bits.
 
-
 ## Specification
 
 The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
 
-*Note* The following specifications use syntax from Solidity `0.8.7` (or above)
+_Note_ The following specifications use syntax from Solidity `0.8.7` (or above)
 
 - Permission and role MUST be defined as an `uint256`
 - Permission MUST be defined as a power of two
 - Permission MUST be unique
 - `0` MUST be used for none permission
 
+```solidity
+pragma solidity ^0.8.7;
+
+/**
+    @title EIP-6617 Bit Based Permission
+    @dev See https://eips.ethereum.org/EIPS/eip-6617
+*/
+library EIP6617 {
```

Comment:
I think my last complaint before merging as draft is that this should be an interface.

Reply:
This part is library that focused on bit manipulation that allow us to compose/discompose an `uint256` permission. In my opinion, it shouldn't be an interface we have no benefit by implement it as an interface.

Reply:
A library that implements bit manipulation doesn't need to be an EIP. It can live on GitHub, as an npm package, or in OpenZeppelin. You can have multiple different bit manipulation libraries that make different choices in implementation.

The EIP should only describe externally visible functionality, and in the case of Solidity that usually means just interfaces.